static/js/compose_ui.js

```diff
@@ -219,6 +219,13 @@ export function handle_keydown(event, $textarea) {
 }
 
 export function handle_keyup(event, $textarea) {
+    // Change compose close button tooltip as per condition.
+    const message_length = $($textarea).val().replace(/\s+/g, "").length;
```

Comment:
I'd prefer to not add a second code path, in addition to the existing `compose_validate.check_overflow_text`, that measures the length of the message on every keypress -- they can be 10k characters, and thus this could be potentially expensive.

I think the right solution might be to have `compose_validate.check_overflow_text` return the message's length (already trimmed, since it reads `compose_state.message_content()`), and then move the couple lines below to extend this block in `compose.js`:

```
    $("#compose-textarea").on("input propertychange", () => {                                            
        compose_validate.warn_if_topic_resolved(false);                                                  
        compose_validate.check_overflow_text();                                                          
    });                                                                                                  
```

Reply:
Updated the PR as your suggestion.